src/main.rs

```diff
@@ -140,26 +142,42 @@ fn show_help() {
     println!("{}", CARGO_CLIPPY_HELP);
 }
 
+#[allow(print_stdout)]
+fn show_version() {
+    println!("{}", env!("CARGO_PKG_VERSION"));
+}
+
+#[cfg_attr(feature = "cargo-clippy", allow(print_stdout))]
 pub fn main() {
     use std::env;
 
     if env::var("CLIPPY_DOGFOOD").map(|_| true).unwrap_or(false) {
         panic!("yummy");
     }
+    
+    // Check for version and help flags even when invoked as 'cargo-clippy'
+    if std::env::args().any(|a| a == "--help" || a == "-h") {
+        show_help();
+        return;
+    }
+    if std::env::args().any(|a| a == "--version" || a == "-V") {
+        show_version();
+        return;
+    }
 
     let dep_path = env::current_dir().expect("current dir is not readable").join("target").join("debug").join("deps");
 
     if let Some("clippy") = std::env::args().nth(1).as_ref().map(AsRef::as_ref) {
         // this arm is executed on the initial call to `cargo clippy`
 
-        if std::env::args().any(|a| a == "--help" || a == "-h") {
-            show_help();
-            return;
-        }
-
         let manifest_path_arg = std::env::args().skip(2).find(|val| val.starts_with("--manifest-path="));
 
-        let mut metadata = cargo::metadata(manifest_path_arg.as_ref().map(AsRef::as_ref)).expect("could not obtain cargo metadata");
+        let mut metadata = if let Ok(metadata) = cargo::metadata(manifest_path_arg.as_ref().map(AsRef::as_ref)) {
+            metadata
+        } else {
+            let _ = io::stderr().write_fmt(format_args!("error: Could not obtain cargo metadata."));
+            process::exit(101);
```

Comment:
we have a pretty random policy with `panic`s vs `process::exit`, so I guess this is fine. What's the reason for this change though?